extensions/sql-migration/src/api/azure.ts

```diff
@@ -285,19 +318,105 @@ export interface StartDatabaseMigrationRequest {
 	}
 }
 
-export interface DatabaseMigration {
-	properties: {
-		name: string,
-		provisioningState: string,
-		sourceDatabaseName: string,
-		migrationOperationId: string,
-	},
-	id: string,
-	name: string,
-	type: string
-}
-
 export interface StartDatabaseMigrationResponse {
 	status: number,
 	databaseMigration: DatabaseMigration
 }
+
+export interface DatabaseMigration {
+	properties: DatabaseMigrationProperties;
+	id: string;
+	name: string;
+	type: string;
+}
+export interface DatabaseMigrationProperties {
+	scope: string;
+	provisioningState: string;
+	migrationStatus: string;
+	migrationStatusDetails?: MigrationStatusDetails;
+	sourceSqlConnection: SqlConnectionInfo;
+	sourceDatabaseName: string;
+	targetDatabaseCollation: string;
+	migrationController: string;
+	migrationOperationId: string;
+	backupConfiguration: BackupConfiguration;
+	autoCutoverConfiguration: AutoCutoverConfiguration;
+	migrationFailureError: ErrorInfo;
+}
+export interface MigrationStatusDetails {
+	migrationState: string;
+	startedOn: string;
+	endedOn: string;
+	fullBackupSetInfo: BackupSetInfo;
+	lastRestoredBackupSetInfo: BackupSetInfo;
+	activeBackupSets: BackupSetInfo[];
+	blobContainerName: string;
+	isFullBackupRestored: boolean;
+	restoreBlockingReason: string;
+	fileUploadBlockingErrors: string[];
+	currentRestoringFileName: string;
+	lastRestoredFilename: string;
+}
+
+export interface SqlConnectionInfo {
```

Comment:
do we really need new connection type?  i.e. we can use on of the existing ones?

Reply:
This is the connection Info object from api definitions. It is only used in azure stuff.

Reply:
why does it need to be redeclared here?